### PR TITLE
chore: move args for `createEncoder` into non-repeating `options` & fix typedoc

### PR DESCRIPTION
**Pull request by danisharora099**
ref: https://github.com/waku-org/js-waku/issues/1010#issuecomment-1409779701

### CODE REVIEW

**Review by weboko — APPROVED** [review]
Looks good

File: packages/core/src/lib/message/version_0.ts [hunk 1]
```diff
@@ -99,14 +100,11 @@ export class Encoder implements IEncoder {
...
  * messages.
- *
- * @param contentTopic The content topic to set on outgoing messages.
- * @param ephemeral An optional flag to mark message as ephemeral, ie, not to be stored by Waku Store nodes.
  */
-export function createEncoder(
-  contentTopic: string,
-  ephemeral = false
-): Encoder {
+export function createEncoder({
```
Comment (fryorcraken) [inline packages/core/src/lib/message/version_0.ts:104]: Do we think we want to accept `string | EncoderOptions` so most of the time, the developer just pass the content topic `string` and only pass an `object` in advance usage?

WDYT @hackyguru ?

File: packages/message-encryption/src/ecies.ts [hunk 2]
```diff
@@ -63,6 +64,13 @@ export class Encoder implements IEncoder {
   }
 }
 
+export interface EciesEncoderOptions extends EncoderOptions {
```
Comment (fryorcraken) [inline packages/message-encryption/src/ecies.ts:67]: IMO we are in the `ecies` package so the `Ecies` prefix is not needed.

File: packages/message-encryption/src/symmetric.ts [hunk 3]
```diff
@@ -61,6 +62,13 @@ export class Encoder implements IEncoder {
   }
 }
 
+export interface SymmetricEncoderOptions extends EncoderOptions {
```
Comment (fryorcraken) [inline packages/message-encryption/src/symmetric.ts:65]: IMO we are in the `symmetric` package so the `Symmetric` prefix is not needed.

**Review by fryorcraken — COMMENTED** [review]
Need to update changelog to document breaking api change

**Pull request merged**